### PR TITLE
[ts-sdk-v2] Add MIME types, and convert input types from string to Enum

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Adds mime types, framework for allowing BCS in transaction submission and in API output.

### Test Plan
`pnpm fmt && pnpm test`

### CODE REVIEW

**Review by heliuchuan — APPROVED** [review]

File: ecosystem/typescript/sdk_v2/src/client/core.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ async function request<Req, Res>(
   const headers: Record<string, string | string[] | undefined> = {
     ...overrides?.HEADERS,
     "x-aptos-client": `aptos-ts-sdk/${VERSION}`,
-    "content-type": contentType ?? "application/json",
+    "content-type": contentType ?? MimeType.JSON.valueOf(),
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk_v2/src/client/core.ts:36]: why we need the `valueOf`?

File: ecosystem/typescript/sdk_v2/src/client/get.ts [hunk 2]
```diff
@@ -54,7 +58,8 @@ export async function get<Req, Res>(options: GetRequestOptions): Promise<AptosRe
       method: "GET",
       originMethod,
       path,
-      contentType,
+      contentType: contentType?.valueOf(),
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk_v2/src/client/get.ts:61]: also here?

**Review by 0xmaayan — APPROVED** [review]

File: ecosystem/typescript/sdk_v2/src/client/core.ts (hunk 1 above)
Comment (gregnazario) [reply ecosystem/typescript/sdk_v2/src/client/core.ts:36]: Does it automatically convert to string?  I don't know how enums work in typescript necessarily

Comment (0xmaayan) [reply ecosystem/typescript/sdk_v2/src/client/core.ts:36]: I think having `MimeTime.JSON` is enough to output the string...?

**Pull request merged**